### PR TITLE
v2: Rename collector to resource

### LINKED ISSUE #980: v2: Rename collector to resources
Rename use of "collector" in user facing things (flags, help text, etc.) to "resources", as the collector type pattern does not exist anymore.

Merging of this will be blocked until after the v1.9.0 release.

### PR DESCRIPTION
**Pull request by Alexander-Carpenter**
**What this PR does / why we need it**: Rename collector to resource since the collet type pattern is no longer used.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #980

### CODE REVIEW

File: docs/cli-arguments.md [hunk 1]
```diff
@@ -46,6 +45,7 @@ Usage of ./kube-state-metrics:
...
       --pod-namespace string                        Name of the namespace of the pod specified by --pod. When set, it is expected that --pod and --pod-namespace are both set. Most likely this should be passed via the downward API. This is used for auto-detecting sharding. If set, this has preference over statically configured sharding. This is experimental, it may be removed without notice.
       --port int                                    Port to expose metrics on. (default 80)
+      --resources string                            Comma-separated list of resources to be enabled. Defaults to "certificatesigningrequests,configmaps,cronjobs,daemonsets,deployments,endpoints,horizontalpodautoscalers,ingresses,jobs,limitranges,mutatingwebhookconfigurations,namespaces,networkpolicies,nodes,persistentvolumeclaims,persistentvolumes,poddisruptionbudgets,pods,replicasets,replicationcontrollers,resourcequotas,secrets,services,statefulsets,storageclasses,validatingwebhookconfigurations,volumeattachments"
```
Comment (tariq1890) [inline docs/cli-arguments.md]:
```suggestion
      --resources string                            Comma-separated list of Kubernetes resources to be enabled. Defaults to "certificatesigningrequests,configmaps,cronjobs,daemonsets,deployments,endpoints,horizontalpodautoscalers,ingresses,jobs,limitranges,mutatingwebhookconfigurations,namespaces,networkpolicies,nodes,persistentvolumeclaims,persistentvolumes,poddisruptionbudgets,pods,replicasets,replicationcontrollers,resourcequotas,secrets,services,statefulsets,storageclasses,validatingwebhookconfigurations,volumeattachments"
```

File: docs/design/metrics-store-performance-optimization.md [hunk 2]
```diff
@@ -83,9 +83,9 @@ the cache map and concatenates all time series string blobs into a single
 string, which is finally passed on as a response.
 
 ```
-       +---------------+ +-----------+                 +---------------+         +-------------------+
-       | pod_reflector | | pod_store |                 | pod_collector |         | metrics_endpoint  |
-       +---------------+ +-----------+                 +---------------+         +-------------------+
+       +---------------+ +-----------+                 +--------------+         +-------------------+
+       | pod_reflector | | pod_store |                 | pod_resource |         | metrics_endpoint  |
```
Comment (tariq1890) [inline docs/design/metrics-store-performance-optimization.md]: I don't think this should be changed.

**Comment by LiliC** [comment]
@Alexander-Carpenter friendly ping in case you missed it :)

**Comment by Alexander-Carpenter** [comment]
Sorry had an issue with rebase so recommited the changes

File: pkg/options/options.go [hunk 3]
```diff
@@ -82,7 +82,7 @@ func (o *Options) AddFlags() {
 	o.flags.StringVar(&o.Host, "host", "0.0.0.0", `Host to expose metrics on.`)
 	o.flags.IntVar(&o.TelemetryPort, "telemetry-port", 8081, `Port to expose kube-state-metrics self metrics on.`)
 	o.flags.StringVar(&o.TelemetryHost, "telemetry-host", "0.0.0.0", `Host to expose kube-state-metrics self metrics on.`)
-	o.flags.Var(&o.Collectors, "collectors", fmt.Sprintf("Comma-separated list of collectors to be enabled. Defaults to %q", &DefaultCollectors))
+	o.flags.Var(&o.Resources, "Resources", fmt.Sprintf("Comma-separated list of Resources to be enabled. Defaults to %q", &DefaultResources))
```
Comment (LiliC) [inline pkg/options/options.go]: This needs to be lowercase, as currently the flag is Resources. After this the tests should pass.
```suggestion
	o.flags.Var(&o.Resources, "resources", fmt.Sprintf("Comma-separated list of Resources to be enabled. Defaults to %q", &DefaultResources))
```

**Pull request merged**